src/components/avatar/avatar.ts

```diff
@@ -0,0 +1,76 @@
+import { html, LitElement } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { Constructor } from '../common/mixins/constructor.js';
+import { SizableMixin } from '../common/mixins/sizable.js';
+import { styles } from './avatar.material.css';
+
+export class IgcAvatarComponent extends SizableMixin(
+  <Constructor<LitElement>>LitElement
+) {
+  static styles = [styles];
+
+  @property()
+  src?: string;
+
+  @state()
+  private hasError = false;
+
+  @property()
+  alt?: string;
+
+  @property()
+  initials?: string;
+
+  @property({ reflect: true })
+  shape: 'circle' | 'rounded' | 'square' = 'square';
+
+  protected get classes() {
+    const { size, shape } = this;
+
+    return {
+      circle: shape === 'circle',
+      rounded: shape === 'rounded',
+      square: shape === 'square',
+      small: size === 'small',
+      medium: size === 'medium',
+      large: size === 'large',
+    };
+  }
+
+  constructor() {
+    super();
+    this.size = 'small';
+  }
+
+  render() {
+    return html`
+      <div
+        part="native"
+        role="img"
+        aria-label="avatar"
+        aria-roledescription=${this.size + ' ' + this.shape}
+        class=${classMap(this.classes)}
+      >
+        ${this.initials
+          ? html`<span part="initials">${this.initials}</span>`
+          : html`
+              <span part="icon">
+                <slot name="icon"></slot>
+              </span>
```

Comment:
We could use the default slot here.
@simeonoff, What do you think?

Reply:
I agree.